src/bsp/src/posix/include/InterlocManager.h

```diff
@@ -1,33 +1,59 @@
 #ifndef __INTERLOCMANAGER_H__
 #define __INTERLOCMANAGER_H__
 
+#include <bsp/IInterlocManager.h>
+#include <gazebo_msgs/ModelStates.h>
+#include <logger/ILogger.h>
+#include <ros/subscriber.h>
+#include <tf2/LinearMath/Transform.h>
+#include <tf2_ros/transform_listener.h>
+
 class InterlocManager : public IInterlocManager {
   public:
+    InterlocManager(ILogger& logger);
     ~InterlocManager() override = default;
 
-    void startInterloc() override{};
+    void startInterloc() override;
 
     void setPositionUpdateCallback(positionUpdateCallbackFunction_t callback,
-                                   void* context) override {
-        // TODO: implement
-        (void)callback;
-        (void)context;
-    };
+                                   void* context) override;
 
+    // Calib API is not needed in simulation. Can just ignore any calls to these functions
```

Comment:
Even if there is no logic here, you could put the definition of the function in the .cpp so the compilation time is reduced.